src/components/inputs/checkboxes/CheckboxTree/CheckboxTree.tsx

```diff
@@ -575,21 +582,35 @@ function createIsLeafVisible<T>(
   getNodeId: CheckboxTreeProps<T>['getNodeId'],
   getNodeChildren: CheckboxTreeProps<T>['getNodeChildren'],
   isAdditionalFilterApplied: CheckboxTreeProps<T>['isAdditionalFilterApplied'],
-  isSearchable: CheckboxTreeProps<T>['isSearchable']
+  isSearchable: CheckboxTreeProps<T>['isSearchable'],
+  filteredList: CheckboxTreeProps<T>['filteredList'] = [],
 ) {
-  // if not searching, then all nodes are visible
-  if (!isActiveSearch(isAdditionalFilterApplied, isSearchable, searchTerm)) {
+  // if not searching and no filtered list is passed in, then all nodes are visible
+  if (!isActiveSearch(isAdditionalFilterApplied, isSearchable, searchTerm) && !filteredList.length) {
     return (nodeId: string) => true;
   }
   // otherwise must construct array of visible leaves
   const visibleLeaves = new Set<string>();
   const searchTerms = parseSearchQueryString(searchTerm);
+  const filteredSet = new Set(filteredList);
   const addVisibleLeaves = (node: T, parentMatches: boolean) => {
-    // if parent matches, automatically match (always show children of matching parents)
-    const nodeMatches = (parentMatches || searchPredicate(node, searchTerms));
+    const nodeId = getNodeId(node);
+    let nodeMatches = false;
+    if (parentMatches) {
+      // if parent matches, automatically match (always show children of matching parents)
+      nodeMatches = parentMatches;
+    } else if (searchTerm && filteredList.length) {
```

Comment:
I think we should distinguish between no `filteredList` and an empty `filteredList`, just in case the upstream filtering produces an empty array. I think it might be wrong to assume that means "show everything". What do you think?

Reply:
Let me know if the changes I made address this concern. In short, I removed the default assignment of an empty array for `filteredList` and amended the early return to assign visibility to all nodes when:

1. no search is active
2. no additional filters are applied
3. filteredList is undefined (as opposed to `[]`)